src/array.jl

```diff
@@ -639,13 +639,23 @@ since it needs to check whether levels are used or not.
 """
 unique(A::CategoricalArray{T}) where {T} = _unique(T, A.refs, A.pool)
 
-"""
-    droplevels!(A::CategoricalArray)
-
-Drop levels which do not appear in categorical array `A` (so that they will no longer be
-returned by [`levels`](@ref)).
-"""
-droplevels!(A::CategoricalArray) = levels!(A, filter!(!ismissing, unique(A)))
+if VERSION >= v"0.7.0-DEV.4882"
+    """
+        droplevels!(A::CategoricalArray)
+
+    Drop levels which do not appear in categorical array `A` (so that they will no longer be
+    returned by [`levels`](@ref)).
+    """
```

Comment:
Could we apply the docstring just once instead of duplicating it? For example,
```julia
if VERSION ...
    droplevels!(x) = bingo
else
    droplevels!(x) = bongo
end

"""
doc string
"""
droplevels!
```

Reply:
We could, but I tend to prefer constructs which Femtocleaner will be able to turn into the most natural code automatically when we drop 0.6 support, even if it's temporarily more verbose.

Reply:
Okay, that's fine.